controllers/rcj_soccer_referee_supervisor/referee/penalty_area_checker.py

```diff
@@ -0,0 +1,73 @@
+from typing import List
+
+from referee.consts import BLUE_PENALTY_AREA, YELLOW_PENALTY_AREA
+
+
+class PenaltyAreaChecker:
+    def __init__(self, allowed: int, reset_after: int, time: int):
+        self.allowed = allowed
+        self.reset_after = reset_after
+        self.time = time
+
+        self.y_vertical, self.y_lower, self.y_upper = YELLOW_PENALTY_AREA
+        self.b_vertical, self.b_lower, self.b_upper = BLUE_PENALTY_AREA
+        self.reset()
+
+    def reset(self):
+        self.time_entered_penalty = -1
+        self.time_left_penalty = -1
+
+    def is_in_yellow_penalty(self, x: int, z: int) -> bool:
+        return x < self.y_vertical and self.y_lower < z < self.y_upper
+
+    def is_in_blue_penalty(self, x: int, z: int) -> bool:
+        return x > self.b_vertical and self.b_lower < z < self.b_upper
+
+    @property
+    def is_outside_of_penalty_for_longer(self) -> bool:
```

Comment:
@Adman this is a linguistics nitpick, but should not this be `was_outside_of_penalty_for_longer`? It seems that at the time of calling we are interested in what happened in the past, not the current state.

Feel free to ignore this though -- it's the ultimate nitpick :)

Reply:
At the time of calling the robot still `is` outside of penalty area when the method returns `True`. So what about  `has_been_outside_penalty_for_longer` ? :D